src/pe/exception.rs

```diff
@@ -0,0 +1,826 @@
+//! Exception handling and stack unwinding for x64.
+//!
+//! Exception information is exposed via the [`ExceptionData`] structure. If present in a PE file,
+//! it contains a list of [`RuntimeFunction`] entries that can be used to get [`UnwindInfo`] for a
+//! particular code location.
+//!
+//! Unwind information contains a list of unwind codes which specify the operations that are
+//! necessary to restore registers (including the stack pointer RSP) when unwinding out of a
+//! function.
+//!
+//! Depending on where the instruction pointer lies, there are three strategies to unwind:
+//!
+//!  1. If the RIP is within an epilog, then control is leaving the function, there can be no
+//!     exception handler associated with this exception for this function, and the effects of the
+//!     epilog must be continued to compute the context of the caller function. To determine if the
+//!     RIP is within an epilog, the code stream from RIP on is examined. If that code stream can be
+//!     matched to the trailing portion of a legitimate epilog, then it's in an epilog, and the
+//!     remaining portion of the epilog is simulated, with the context record updated as each
+//!     instruction is processed. After this, step 1 is repeated.
+//!
+//!  2. Case b) If the RIP lies within the prologue, then control has not entered the function,
+//!     there can be no exception handler associated with this exception for this function, and the
+//!     effects of the prolog must be undone to compute the context of the caller function. The RIP
+//!     is within the prolog if the distance from the function start to the RIP is less than or
+//!     equal to the prolog size encoded in the unwind info. The effects of the prolog are unwound
+//!     by scanning forward through the unwind codes array for the first entry with an offset less
+//!     than or equal to the offset of the RIP from the function start, then undoing the effect of
+//!     all remaining items in the unwind code array. Step 1 is then repeated.
```

Comment:
*step 2?

Reply:
No, it's step 1 from https://docs.microsoft.com/en-us/cpp/build/exception-handling-x64?view=vs-2017#unwind-procedure (the 1,2,3 here are actually case a,b,c from that doc).